gateway/gateway.go

```diff
@@ -1479,6 +1471,39 @@ func (gateway *HandleT) StartAdminHandler(ctx context.Context) error {
 	return g.Wait()
 }
 
+func (gateway *HandleT) maxActiveClientMiddleware(next http.Handler) http.Handler {
+	atomic.AddInt32(&gateway.activeClientCount, 1)
+	defer func() {
+		atomic.AddInt32(&gateway.activeClientCount, -1)
+	}()
+
+	//this is to make sure that we don't have more than `maxClient` in-memory at any point of time. As, having more http client than `maxClient`
+	// may lead to gateway OOM kill.
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if maxActiveClients != 0 {
+			select {
+			case gateway.activeClient <- struct{}{}:
+				defer func() {
+					<-gateway.activeClient
+				}()
+				next.ServeHTTP(w, r)
+			default:
+				http.Error(w, "server is under pressure", response.GetStatusCode(response.TooManyRequestsInMemory))
```

Comment:
```suggestion
				http.Error(w, http.StatusText(http.StatusServiceUnavailable), http.StatusServiceUnavailable)
```

do we have any specific reason not to prefer using the standard `net/http` status and messages?

Reply:
actually, for rest of the place in gateway code I saw that we have defined a response package & we were using it to respond. So, I also did the same way. But, just noticed that we only calling `response.GetStatus` in gateway and not `response.GetStatusCode`. So, modified it.